core-bundle/src/Resources/contao/library/Contao/Email.php

```diff
@@ -120,13 +124,21 @@ class Email
 	/**
 	 * Instantiate the object and load the mailer framework
 	 *
-	 * @param \Swift_Mailer|null $objMailer
+	 * @param \Swift_Mailer|MailerInterface|null $objMailer
 	 */
-	public function __construct(\Swift_Mailer $objMailer = null)
+	public function __construct($objMailer = null)
 	{
-		$this->objMailer = $objMailer ?: System::getContainer()->get('swiftmailer.mailer');
+		$this->objMailer = $objMailer ?: System::getContainer()->get('mailer');
 		$this->strCharset = Config::get('characterSet');
-		$this->objMessage = new \Swift_Message();
+
+		if ($this->objMailer instanceof MailerInterface)
+		{
+			$this->objMessage = new EmailMessage();
+		}
+		else
+		{
+			$this->objMessage = new \Swift_Message();
+		}
```

Comment:
Should we throw an exception if `$objMailer` is neither?

Reply:
Sure, why not :)